modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFClaimForm/components/MDFClaimPage/schema/yup.ts

```diff
@@ -169,11 +141,138 @@ const claimSchema = object({
 								}
 							),
 				}),
-				metrics: string().max(
-					350,
-					'You have exceeded the character limit'
+				listOfQualifiedLeads: mixed()
+					.when('selected', {
+						is: (selected: boolean) => selected,
+						then: (schema) =>
+							schema
+								.test(
+									'fileSize',
+									validateDocument.fileSize.message,
+									(listOfQualifiedLeads) => {
+										if (
+											listOfQualifiedLeads &&
+											!listOfQualifiedLeads.id
+										) {
+											return listOfQualifiedLeads
+												? Math.ceil(
+														listOfQualifiedLeads.size /
+															1000
+												  ) <=
+														validateDocument
+															.fileSize.maxSize
+												: false;
+										}
+
+										return true;
+									}
+								)
+								.test(
+									'fileType',
+									validateDocument.listOfLeadsDocuments
+										.message,
+									(listOfQualifiedLeads) => {
+										if (
+											listOfQualifiedLeads &&
+											!listOfQualifiedLeads.id
+										) {
+											return listOfQualifiedLeads
+												? validateDocument.listOfLeadsDocuments.types.includes(
+														listOfQualifiedLeads.type
+												  )
+												: false;
+										}
+
+										return true;
+									}
+								),
+					})
+					.when(['selected', 'typeActivity'], {
+						is: (
+							selected: boolean,
+							typeActivity: LiferayPicklist
+						) =>
+							(selected &&
+								typeActivity.key === TypeActivityKey.EVENT) ||
+							(selected &&
+								typeActivity.key ===
+									TypeActivityKey.MISCELLANEOUS_MARKETING),
```

Comment:
```suggestion
							(
								typeActivity.key === TypeActivityKey.EVENT) ||
								typeActivity.key ===
									TypeActivityKey.MISCELLANEOUS_MARKETING) && selected,
```

Reply:
if this logic is the same from the component, it's better to centralize in a function